### PR TITLE
add suricata project

### PR DESCRIPTION
**Pull request by koltiradw**

### CODE REVIEW

File: projects/suricata/build.sh [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+cd fuzzpcap
+rm -rf build
+mkdir build
+cd build
+cmake ..
+make install
+)
+
+(
+cp -r libhtp suricata && cd suricata
```
Comment (apach301) [inline projects/suricata/build.sh:127]: `rm -rf suricata/libhtp` before copying. (But anyway libhtp is cleaned during suricata's `make clean`)

File: projects/suricata/build.sh [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+make clean
+)
+
+unset CFLAGS
+unset CXXFLAGS
+unset LDFLAGS
+unset RUSTFLAGS
+
+echo "[x] Sydr stage"
+export CC=clang
```
Comment (apach301) [inline projects/suricata/build.sh:150]: CXX, CFLAGS, CXXFLAGS?

File: projects/suricata/fuzz_applayerprotodetectgetproto-afl++.toml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+exit-on-time = 7200
+
+[sydr]
+args = "-l trace -s 180"
+target = "/SRC/suricata/sydr/fuzz_applayerprotodetectgetproto @@"
+jobs = 1
+
+[[aflplusplus]]
+args = "-t 10000000 -i /corpus_applayerprotodetectgetproto/ -x /suricata.dict"
```
Comment (apach301) [inline projects/suricata/fuzz_applayerprotodetectgetproto-afl++.toml:25]: Missing suricata.dict file

File: projects/suricata/fuzz_applayerprotodetectgetproto.toml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+exit-on-time = 7200
+
+[sydr]
+args = "-l trace -s 180"
+target = "/SRC/suricata/sydr/fuzz_applayerprotodetectgetproto @@"
+jobs = 1
+
+[libfuzzer]
+path = "/SRC/suricata/afl/fuzz_applayerprotodetectgetproto"
```
Comment (apach301) [inline projects/suricata/fuzz_applayerprotodetectgetproto.toml:25]: libfuzzer

File: projects/suricata/fuzz_mimedecparseline.toml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+exit-on-time = 7200
+
+[sydr]
+args = "-l trace -s 180"
+target = "/SRC/suricata/sydr/fuzz_mimedecparseline @@"
+jobs = 1
+
+[libfuzzer]
+path = "/SRC/suricata/afl/fuzz_mimedecparseline"
```
Comment (apach301) [inline projects/suricata/fuzz_mimedecparseline.toml:25]: libfuzzer

File: projects/suricata/fuzz_predefpcap_aware.toml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+exit-on-time = 7200
+
+[sydr]
+args = "-l trace -s 180"
+target = "/SRC/suricata/sydr/fuzz_predefpcap_aware @@"
+jobs = 1
+
+[libfuzzer]
+path = "/SRC/suricata/afl/fuzz_predefpcap_aware"
```
Comment (apach301) [inline projects/suricata/fuzz_predefpcap_aware.toml:25]: libfuzzer

File: projects/suricata/fuzz_sigpcap_aware.toml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+exit-on-time = 7200
+
+[sydr]
+args = "-l trace -s 180"
+target = "/SRC/suricata/sydr/fuzz_sigpcap_aware @@"
+jobs = 1
+timeout = 1800
+
+[libfuzzer]
+path = "/SRC/suricata/afl/fuzz_sigpcap_aware"
```
Comment (apach301) [inline projects/suricata/fuzz_sigpcap_aware.toml:26]: libfuzzer

**Review by apach301 — CHANGES_REQUESTED** [review]

File: projects/suricata/build.sh [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+cd fuzzpcap
+rm -rf build
+mkdir build
+cd build
+cmake ..
+make install
+)
+
+(
+sh autogen.sh
```
Comment (apach301) [inline projects/suricata/build.sh:112]: cd suricata

File: projects/suricata/build.sh [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+cd fuzzpcap
+rm -rf build
+mkdir build
+cd build
+cmake ..
+make install
+)
+
+(
+sed -i 's/fuzz_applayerparserparse.c/sydr_applayerparserparse.c/g' src/Makefile.am
```
Comment (apach301) [inline projects/suricata/build.sh:159]: cd suricata

File: projects/suricata/build.sh [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+cd fuzzpcap
+rm -rf build
+mkdir build
+cd build
+cmake ..
+make install
+)
+
+(
+sh autogen.sh
```
Comment (apach301) [inline projects/suricata/build.sh:218]: cd suricata

File: projects/suricata/build.sh (hunk 2 above)
Comment (apach301) [reply projects/suricata/build.sh:129]: There were CXX, LDFLAGS, and RUSTFLAGS set during previous build steps. Could they affect sydr-build?

**Review by apach301 — CHANGES_REQUESTED** [review]

File: projects/suricata/build.sh (hunk 2 above)
Comment (koltiradw) [reply projects/suricata/build.sh:130]: yep, my fault

File: projects/suricata/fuzz_applayerprotodetectgetproto-afl++.toml [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+################################################################################
+
+exit-on-time = 7200
+
+[sydr]
+args = "-l trace -s 180"
+target = "/SRC/suricata/sydr/fuzz_applayerprotodetectgetproto @@"
+jobs = 1
+
+[[aflplusplus]]
```
Comment (kobrineli) [inline projects/suricata/fuzz_applayerprotodetectgetproto-afl++.toml:24]: change `[[` to `[`

Comment (kobrineli) [reply projects/suricata/fuzz_applayerprotodetectgetproto-afl++.toml:24]: here and in the other `toml`

**Review by kobrineli — CHANGES_REQUESTED** [review]

File: projects/suricata/fuzz_applayerprotodetectgetproto-afl++.toml (hunk 11 above)
Comment (koltiradw) [reply projects/suricata/fuzz_applayerprotodetectgetproto-afl++.toml]: ty!

File: projects/suricata/build.sh [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+cp src/fuzz_applayerprotodetectgetproto libfuzzer/fuzz_applayerprotodetectgetproto
+cp src/fuzz_mimedecparseline libfuzzer/fuzz_mimedecparseline
+cp src/fuzz_predefpcap_aware libfuzzer/fuzz_predefpcap_aware
+cp src/fuzz_sigpcap_aware libfuzzer/fuzz_sigpcap_aware
+
+make clean
+)
+
+echo "[x] AFL++ stage"
+export CC=afl-clang-lto
```
Comment (apach301) [inline projects/suricata/build.sh:78]:
```suggestion
export CC=afl-clang-fast
```

File: projects/suricata/build.sh [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+export CXX=afl-clang-lto++
```
Comment (apach301) [inline projects/suricata/build.sh:79]:
```suggestion
export CXX=afl-clang-fast++
```

**Review by apach301 — CHANGES_REQUESTED** [review]

**Review by apach301 — APPROVED** [review]

**Review by kobrineli — APPROVED** [review]

**Comment by koltiradw** [comment]
change lto to fast may affect exec speed for pcap-aware targets. @apach301  FYI.

**Pull request merged**